### PR TITLE
Support `nan`, `posinf`, `neginf` in `cupy.nan_to_num`

### LINKED ISSUE #4867: Supporting nan_to_num's "nan", "posinf", and "neginf" arguments
[cupy.nan_to_num](https://docs.cupy.dev/en/stable/reference/generated/cupy.nan_to_num.html) was implemented in #1391, but as a ufunc instead of a function that takes non-array arguments, as in NumPy. This means we can't pass the `nan`, `posinf`, or `neginf` arguments to it, as in [numpy.nan_to_num](https://docs.scipy.org/doc/numpy/reference/generated/numpy.nan_to_num.html).

My CuPy version is 8.5.0 (updated to the latest on conda-forge).

### PR DESCRIPTION
**Pull request by eternalphane**
Fixes #4867.

### CODE REVIEW

File: cupy/_math/misc.py [hunk 1]
```diff
@@ -316,46 +316,119 @@ def clip(a, a_min=None, a_max=None, out=None):
...
+
+    .. seealso:: :func:`numpy.nan_to_num`
 
     ''')
 
 
+def nan_to_num(x, out=None, *, nan=0.0, posinf=None, neginf=None, **kwds):
+    """ Elementwise nan_to_num function.
+
+    .. seealso:: :func:`numpy.nan_to_num`
```
Comment (kmaehashi) [inline cupy/_math/misc.py:424]: Thanks! This also fixes the deadlink 😄 
https://docs.cupy.dev/en/latest/reference/generated/cupy.nan_to_num.html

**Review by takagi — DISMISSED** [review]

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
I'll add tests for the parameters in another PR.

**Review by takagi — DISMISSED** [review]

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
pfnCI, test this please.

**Review by takagi — APPROVED** [review]

**Comment by takagi** [comment]
Thank you!

**Pull request merged**